Add group role improvements doc (#2603)
wso2/docs-is: en/docs/setup/migrating-what-has-changed.md

![relationship-between-groups-and-roles](../../assets/img/setup/groups-roles-relationship.png)

> ## Group and role separation Improvements

Reviewer: ```suggestion
## Group and role separation improvements
```

Identity server 5.11.0 introduced group and role separation. 

However, only certain flows were making use of this separation, and it was not adopted throughout 
the product flows. This release introduces changes to the product flows to adapt this separation.

### Abbreviations

* Wso2.role claim = http://wso2.org/claims/role
* Wso2.roles claim = http://wso2.org/claims/roles
* Wso2.groups claim = http://wso2.org/claims/groups

### List of improvements

Following are the improvements introduced with this effort.

#### Carbon kernel level support

> getUserClaimValues(), getUsersClaimValues(), getUserClaimValuesWithID(), getUsersClaimValuesWithID() 

Reviewer: ```suggestion
`getUserClaimValues()`, `getUsersClaimValues()`, `getUserClaimValuesWithID()`, `getUsersClaimValuesWithID()` 
```

are the APIs from carbon kernel which provide claim values of users at the OSGi level. Previously 
these APIs specifically treated the wso2.role claim so that it would return both roles and groups. 

> Now in addition to that, both wso2.groups and wso2.roles(A new claim introduced with this effort) 

Reviewer: ```suggestion
Now in addition to that, both wso2.groups and wso2.roles (a new claim introduced with this effort) 
```

claims return groups and roles respectively.

> #### Going forward, discouraging the usages of wso2.role claim

Reviewer: ```suggestion
#### The usage of wso2.role claim going forward.
```

Wso2.role claim is no longer encouraged to be used in configurations(such as role mapping, 
requested claims, etc), and with extensions. Therefore, following changes are made regarding the 
wso2.role claim.

* No longer a  ‘supported by default’ claim
* Hidden from listing local claims(local claims section, requested claims section in service 
  providers, adding/editing external claims section, etc)
* Hidden from the user profile

The following claims are the alternatives that must be utilized instead.

* Wso2.roles claim
  * A newly introduced claim URI.
  * Treated similar to the wso2.role claim(read-only, supported by default, etc)
  * Contain Internal roles of a user
* Wso2.groups claim
  * Proper utilization of an existing claim URI.
  * Treated similar to the wso2.role claim(read-only, supported by default, etc).
  * Contain groups of a user
  

> #### Service provider role mapping and IdP role mapping  are restricted to roles

Reviewer: ```suggestion
#### Service provider role mapping and identity provider role mapping restrictions
```
